libsolidity/ast/Types.cpp

```diff
@@ -740,6 +740,18 @@ TypeResult IntegerType::binaryOperatorResult(Token _operator, Type const* _other
 		return this;
 	}
 
+	if (auto const* other = dynamic_cast<RationalNumberType const*>(_other); other)
+	{
+		if (_operator == langutil::Token::Mod || _operator == langutil::Token::Div)
+		{
+			std::optional<rational> value = ConstantEvaluator::evaluateBinaryOperator(_operator, {0}, other->value());
```

Comment:
Why not just check `if(other->value().denominator() == 0)`?

Reply:
yeah that's fair :D good catch